### DISCUSSION TITLE
How to define enumeration with int value in IDL?

### DISCUSSION THREAD

**Discussion by Octavian-Zhang** [eProsima/Fast-DDS-Gen#221] [Q&A]
Is it possible to define an enumeration as the following in IDL?
```
enum Enumeration : int32 
{
    RED = 1,
    GREEN = 2,
    BLUE = 3
};
```
If yes, what's the correct syntax for such a definition? The code snippet above isn't valid in IDL.

**Comment by JLBuenoLopez** [comment]
@Octavian-Zhang 

The [IDL to C++11 language mapping](https://www.omg.org/spec/CPP11/1.6/Beta1/About-CPP11) specification states in clause 6.9 that an IDL enumeration should map to an `uint32_t` C++11 enumeration. The [DDS XTypes specification](https://www.omg.org/spec/DDS-XTypes) allows to modify the size of the enumerated type using the `@bit_bound` builtin annotation (Fast DDS-Gen does not support yet this specific annotation for enumerations, as stated in [Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/v2.11.2/fastddsgen/dataTypes/dataTypes.html#annotations)). It does not seem to be a manner to define negative values within an IDL enumeration.
